be/src/vec/functions/array/function_array_lambda_index.cpp

```diff
@@ -45,11 +45,12 @@ class FunctionContext;
 namespace doris::vectorized {
 
 // array_first_index([0, 1, 0]) -> [2]
-class FunctionArrayFirstIndex : public IFunction {
+template <bool first>
+class FunctionArrayLambdaIndex : public IFunction {
 public:
```

Comment:
if this class is just for `array_first_index` `array_last_index`, so i think this class could be renamed to `FunctionArrayFirstOrLastIndex` is better?  Or we need some enum